mlflow/tracking/_tracking_service/client.py

```diff
@@ -231,14 +234,22 @@ def search_traces(
             order_by=order_by,
             page_token=page_token,
         )
+
+        def fn(trace_info: trace_info) -> Optional[TraceData]:
+            try:
+                trace_data = self._download_trace_data(trace_info.request_id)
+            except requests.exceptions.HTTPError as e:
+                if e.response.status_code == 404:
```

Comment:
What if the trace data is malformed? E.g. because the file was only partially written. We should have some sort of exception that we can catch for that too & add a test.